### PR TITLE
docs(nvim_set_keymap): specify that optional arguments defaults to false

### LINKED ISSUE #16919: Default values for 'opts' in nvim_set_keymap ?
### Feature already in Vim?

_No response_

### Feature description

The docs for `:h nvim_set_keymap()` doesn't specify what the default value for opts is. Shouldn't it specify that all values are `false` by default or something ?

This might be pointless now that #16591 is being worked on.

### PR DESCRIPTION
**Pull request by dundargoc**
Closes: https://github.com/neovim/neovim/issues/16919

### CODE REVIEW

**Review by justinmk — APPROVED** [review]
usually preferable to mention the default pretty near the start of the description.

**Comment by dundargoc** [comment]
Sure. Gimme a sec.

File: src/nvim/api/vim.c [hunk 1]
```diff
@@ -1597,7 +1597,7 @@ ArrayOf(Dictionary) nvim_get_keymap(uint64_t channel_id, String mode)
 ///               "desc" can be used to give a description to keymap.
 ///               When called from Lua, also accepts a "callback" key that takes
 ///               a Lua function to call when the mapping is executed.
-///               Values are Booleans. Unknown key is an error.
+///               Values are Booleans (default false). Unknown key is an error.
```
Comment (justinmk) [inline src/nvim/api/vim.c:1600]: e.g. :  

    /// @param  opts  Optional parameters map: keys are |:map-arguments|, values are booleans (default false). Accepts all |:map-arguments|

**Review by justinmk — APPROVED** [review]

File: runtime/doc/api.txt [hunk 2]
```diff
@@ -1593,13 +1593,15 @@ nvim_set_keymap({mode}, {lhs}, {rhs}, {*opts})             *nvim_set_keymap()*
...
-                            to give a description to keymap. When called from
-                            Lua, also accepts a "callback" key that takes a
-                            Lua function to call when the mapping is executed.
-                            Values are Booleans. Unknown key is an error.
+                    {opts}  Optional parameters map: keys are
+                            |:map-arguments|, values are booleans (default
+                            false). Accepts all |:map-arguments| as keys
+                            excluding |<buffer>| but including |noremap| and
+                            "desc". Unknown key is an error. "desc" can be
+                            used to give a description to keymap. When called
```
Comment (zeertzjq) [inline runtime/doc/api.txt:1601]:
```suggestion
                            used to give a description to the mapping. When called
```

File: src/nvim/api/vim.c [hunk 3]
```diff
@@ -1592,12 +1592,13 @@ ArrayOf(Dictionary) nvim_get_keymap(uint64_t channel_id, String mode)
...
-///               When called from Lua, also accepts a "callback" key that takes
-///               a Lua function to call when the mapping is executed.
-///               Values are Booleans. Unknown key is an error.
+/// @param  opts  Optional parameters map: keys are |:map-arguments|, values
+///               are booleans (default false). Accepts all |:map-arguments| as
+///               keys excluding |<buffer>| but including |noremap| and "desc".
+///               Unknown key is an error. "desc" can be used to give a
+///               description to keymap. When called from Lua, also accepts a
+///               "callback" key that takes a Lua function to call when the
+///               mapping is executed.
```
Comment (zeertzjq) [inline src/nvim/api/vim.c:1601]:
```suggestion
///               description to the mapping. When called from Lua, also
///               accepts a "callback" key that takes a Lua function to
///               call when the mapping is executed.
```

**Pull request merged**